src/_pytest/skipping.py

```diff
@@ -208,7 +207,7 @@ def evaluate_xfail_marks(item: Item) -> Optional[Xfail]:
     """Evaluate xfail marks on item, returning Xfail if triggered."""
     for mark in item.iter_markers(name="xfail"):
         run = mark.kwargs.get("run", True)
-        strict = mark.kwargs.get("strict", item.config.getini("xfail_strict"))
+        strict = mark.kwargs.get(item.config.getini("xfail_strict"), True)
```

Comment:
in order to ensure backward compatibility we cannot go straight from  non-strict to strict
instead we have to start by warning if strict was not set to true or false

the warning should indicate that a future major release of pytest will change the default from False to True
and recommend to use struct=True as default and a plugin for actually flaky tests

Reply:
Sure